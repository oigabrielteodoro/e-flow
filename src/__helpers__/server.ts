import { setupServer } from 'msw/node'
import { queryCache } from 'client'

export const server = setupServer()

beforeAll(() => server.listen())
afterEach(() => {
  server.resetHandlers()
  queryCache.clear()
})
afterAll(() => server.close())
