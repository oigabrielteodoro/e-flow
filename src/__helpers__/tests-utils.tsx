import React, { FC, ReactElement } from 'react'
import { render as renderComponent } from '@testing-library/react'

import { Provider } from 'react-redux'
import { ToastContainer } from 'react-toastify'
import { QueryClientProvider } from 'react-query'

import { pipe } from 'fp-ts/function'
import { map, mapLeft, TaskEither } from 'fp-ts/TaskEither'

import { queryClient, store } from 'client'

export function unsafe<T>(value: unknown): T {
  return value as T
}

type Callback = (result: unknown) => unknown

type MapAll = (
  fn: Callback,
) => (data: TaskEither<unknown, unknown>) => TaskEither<unknown, unknown>

export const mapAll: MapAll = (fn) => (data) => {
  return pipe(data, map(fn), mapLeft(fn))
}

export function getErrorMessage(errors: unknown): string {
  return Array.isArray(errors) ? errors[0].message : ''
}

export const AppWrapper: FC = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <Provider store={store}>{children}</Provider>

    <ToastContainer />
  </QueryClientProvider>
)

export const render = (children: ReactElement) => {
  return renderComponent(children, {
    wrapper: AppWrapper,
  })
}
