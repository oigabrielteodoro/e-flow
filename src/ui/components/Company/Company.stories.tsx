import { Story, Meta } from '@storybook/react/types-6-0'

import { Company, Props } from '.'

export default {
  title: 'Company',
  component: Company,
  args: {
    symbol: 'MSFT',
  },
  argTypes: {
    symbol: {
      type: 'string',
    },
    disableFavorite: {
      type: 'boolean',
    },
    as: {
      control: {
        type: 'select',
        options: ['li', 'div'],
      },
      defaultValue: 'div',
    },
  },
} as Meta

export const Basic: Story<Props> = (args) => (
  <div style={{ padding: 10, width: 302 }}>
    <Company {...args} />
  </div>
)
