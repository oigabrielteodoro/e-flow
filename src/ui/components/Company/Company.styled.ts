import styled, { css } from 'styled-components'
import { theme } from 'ui/styles'

type ContainerProps = {
  isPricingUp?: boolean
}

export const Container = styled.div<ContainerProps>`
  display: flex;
  align-items: center;
  padding: 1.6rem;
  border-radius: 0.8rem;
  background: ${theme.colors.white};
  transition: all 0.2s;
  cursor: pointer;

  &:hover {
    transform: translateY(-0.5rem);
    box-shadow: 0 0.8rem 2rem -0.2rem rgba(43, 37, 63, 0.1);
  }

  button {
    background: transparent;
    border: 0;

    & + .logo {
      margin-left: 1rem;
    }
  }

  section {
    display: flex;
    flex-direction: column;
    margin-left: 1.2rem;

    span {
      color: ${theme.colors.gray[900]};

      &:last-child {
        color: ${theme.colors.gray[500]};
      }
    }
  }

  .logo {
    width: 4rem;
    height: 4rem;
    border: 0.2rem solid ${theme.colors.gray[300]};
    border-radius: 50%;
  }

  .pricing {
    margin-left: auto;
    display: flex;
    align-items: center;

    strong {
      font-weight: 500;
      margin-right: 0.8rem;
      color: ${theme.colors.red[500]};

      ${({ isPricingUp }) =>
        isPricingUp &&
        css`
          color: ${theme.colors.green[500]};
        `}
    }
  }
`
