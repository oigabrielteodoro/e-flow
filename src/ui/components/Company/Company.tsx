import React from 'react'

export function Company() {
  return <h1>Company</h1>
}
