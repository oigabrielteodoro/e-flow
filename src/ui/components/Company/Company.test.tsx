import React from 'react'
import { render } from '__helpers__'

import { Company } from './Company'

describe('<Company />', () => {
  it('should render correctly', () => {
    const { container } = render(<Company symbol='MSFT' />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
