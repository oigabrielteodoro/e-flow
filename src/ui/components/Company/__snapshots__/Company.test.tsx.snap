// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Company /> should render correctly 1`] = `
.c1 {
  -webkit-animation: ePEqAU 1.2s ease-in-out infinite;
  animation: ePEqAU 1.2s ease-in-out infinite;
  background-image: linear-gradient( 90deg,rgba(0,0,0,0),#fff,rgba(0,0,0,0) );
  background-color: #b3b3b3;
  background-size: 20rem 100%;
  background-repeat: no-repeat;
  border-radius: 0.313rem;
  display: inline-block;
  line-height: 1;
  opacity: 0.1;
  width: 100%;
  height: 2rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1.6rem;
  border-radius: 0.8rem;
  background: #ffffff;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
}

.c0:hover {
  -webkit-transform: translateY(-0.5rem);
  -ms-transform: translateY(-0.5rem);
  transform: translateY(-0.5rem);
  box-shadow: 0 0.8rem 2rem -0.2rem rgba(43,37,63,0.1);
}

.c0 button {
  background: transparent;
  border: 0;
}

.c0 button + .logo {
  margin-left: 1rem;
}

.c0 section {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 1.2rem;
}

.c0 section span {
  color: #14171a;
}

.c0 section span:last-child {
  color: #657786;
}

.c0 .logo {
  width: 4rem;
  height: 4rem;
  border: 0.2rem solid #acacac;
  border-radius: 50%;
}

.c0 .pricing {
  margin-left: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .pricing strong {
  font-weight: 500;
  margin-right: 0.8rem;
  color: #d64b45;
}

<div
  class="c0"
>
  <button>
    <img
      alt="Icon Star"
      src="/img/icon-star-outline.svg"
    />
  </button>
  <img
    alt="MSFT"
    class="logo"
    src="https://storage.googleapis.com/iex/api/logos/MSFT.png"
  />
  <section>
    <div
      class="c1"
      color="#fff"
      height="2rem"
      width="100%"
    />
    <div
      class="c1"
      color="#fff"
      height="2rem"
      width="100%"
    />
  </section>
  <div
    class="pricing"
  >
    <strong />
    <img
      alt="Icon Pricing"
      src="/img/icon-down-pricing.svg"
    />
  </div>
</div>
`;
